File: app/src/main/java/com/stan/dagger_android/home/MainActivityPresenter.java
package com.stan.dagger_android.home;

import javax.inject.Inject;

/**
 * Created by uu on 2017/12/29.
 */

public class MainActivityPresenter implements MainActivityContract.Presenter {

    MainActivityContract.View view;

    @Inject
    public MainActivityPresenter(){
    }

    @Override
    public void takeView(MainActivityContract.View view) {
        this.view = view;
    }

    @Override
    public void dropView() {
        view = null;
    }

    @Override
    public boolean isDataMissing() {
        return true;
    }
}

File: app/src/main/java/com/stan/dagger_android/home/MainActivityPresenter.kt
package com.stan.dagger_android.home

import com.stan.dagger_android.base.DisposablePresenter
import javax.inject.Inject

/**
 * Created by uu on 2017/12/29.
 */

class MainActivityPresenter @Inject constructor() :DisposablePresenter(), MainActivityContract.Presenter {

    internal var view: MainActivityContract.View? = null

    override fun takeView(view: MainActivityContract.View) {
        this.view = view
    }

    override fun dropView() {
        clearDisposable()
        view = null
    }
}
